csrc/scheduler/utils.cpp

```diff
@@ -1216,6 +1216,24 @@ std::vector<TensorView*> cacheInputs(Fusion* fusion, bool unroll) {
     for (auto use : tv->uses()) {
       if (!use->isA<PadOp>()) {
         cached_uses.push_back(use);
+      } else {
+        // propagate allocation domain on pad op
+        if (!use->as<PadOp>()->out()->isA<TensorView>()) {
+          continue;
+        }
+        auto pad_out_tv = use->as<PadOp>()->out()->as<TensorView>();
+        if (pad_out_tv->hasAllocation()) {
+          continue;
```

Comment:
I wonder if we can really just continue here. What if the pad output has an allocation order that's not the same as the input?

Reply:
Technically we shouldn't. We likely will ended up with a failure after this. I thought about overwriting this but backed out from that.
My old thought is that: 1) If `pad_out_tv` is an intermediate tensor, this means the scheduler or something else messed up vectorization and I intended to throw the error during validation.
2) If `pad_out_tv` is the fusion output, then vectorization analysis should have validated input/output allocation domain already and they must match for vectorization to be enabled. So we should be good.

I think a cleaner alternative is the approach suggested earlier, if we add another set after the pad to restore the original allocation domain, we wouldn't alter any user behavior.

Reply:
Let's have the assumptions as code assertions. Looks like we should have something like:

```
if (pad_out_tv->hasAllocation()) {
  // Comment on why this assertion should hold
  NVF_ERROR(pad_out_tv->isFusionOutput());
}
```